shopfloor/actions/search.py

```diff
@@ -14,13 +14,14 @@ class SearchAction(Component):
 
     # TODO: these methods shall be probably replaced by scan anything handlers
 
-    def location_from_scan(self, barcode):
+    def location_from_scan(self, barcode, limit=1):
         model = self.env["stock.location"]
         if not barcode:
             return model.browse()
-        return model.search(
-            ["|", ("barcode", "=", barcode), ("name", "=", barcode)], limit=1
-        )
+        res = model.search([("barcode", "=", barcode)], limit=limit)
+        if len(res) < limit:
```

Comment:
I don't get why you need to play w/ limit :thinking:

Reply:
This was a requirement, @jbaudoux should know the underlying reason.

Reply:
@jbaudoux ?
I guess the idea is: if we have a limit of 10 and we find only 3 results, we do another search by barcode to find the rest.
Having a comment on top would be nice.

Reply:
The concern is that the search is not scoped to scenario/menu context. So to have a proper match, you must ensure location uniqueness which is not a standard odoo requirement.
Ideally we should be able to pass a domain & limit in the search. Having the limit is a first good improvement, you can then apply scope on the returned recordset

Reply:
Having a comment on top is not only nice, it's a requirement I would say :)

Reply:
Comment added